examples/tonic-key-value-store/src/main.rs

```diff
@@ -174,20 +171,15 @@ async fn serve_forever(listener: TcpListener) -> Result<(), Box<dyn std::error::
         .layer(
             TraceLayer::new_for_grpc().make_span_with(DefaultMakeSpan::new().include_headers(true)),
         )
-        // Build our final `Service`
-        .service(service);
+        .into_inner();
 
-    // Run the service using hyper
+    // Build and run the server
     let addr = listener.local_addr()?;
-
     tracing::info!("Listening on {}", addr);
-
-    // We cannot use `tonic::transport::Server` directly as it requires services to implement
-    // `tonic::transport::NamedService` which tower-http middleware don't
-    Server::from_tcp(listener)?
-        // Required for gRPC
-        .http2_only(true)
-        .serve(Shared::new(service))
+    tonic::transport::Server::builder()
+        .layer(layer)
+        .add_service(service)
+        .serve(addr)
```

Comment:
ah this wont work. Have to use `serve_with_incoming` since the listener has already been create. Fixing now.

Reply:
Fixed in 4192b353eede8